pkg/odo/cli/component/update.go

```diff
@@ -73,16 +66,43 @@ func (uo *UpdateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 // Validate validates the update parameters
 func (uo *UpdateOptions) Validate() (err error) {
+
+	uo.doesComponentExist, err = component.Exists(uo.Context.Client, uo.localConfigInfo.GetName(), uo.localConfigInfo.GetApplication())
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if component of name %s exists in application %s", uo.localConfigInfo.GetName(), uo.localConfigInfo.GetApplication())
```

Comment:
I'm wondering when this error is thrown. I did below but didn't face this error:
1. Created a component with `odo create`
2. Before pushing it, I did `odo update`.

So in this case, component doesn't exist in the application yet, right? At least not on the OpenShift cluster. Am I right in expecting this error in this scenario? :thinking:

Reply:
actually if the component doesn't exist then `doesComponentExist` become false but the error is thrown when maybe connection to the cluster timeout or we dont have permissions to check if the component exists or not

Reply:
Makes sense about the `false` thing. But I still don't hit this error when I stopped my crc cluster. Instead, after building from the latest bits, I get the no such directory error which is covered in the `Validate` function but after this particular piece:
```sh
$ odo update --local ./code
 ✗  error while validating source path: stat ./code: no such file or directory
```

Reply:
Not sure how to forcefully get that error :sweat_smile: